src/Agent.Worker/ContainerOperationProvider.cs

```diff
@@ -606,20 +606,54 @@ private async Task StartContainerAsync(IExecutionContext executionContext, Conta
                         useHostGroupId = true;
                     }
 
-                    if (string.IsNullOrEmpty(containerUserName)) // Create a new user with same UID as on the host
+                    // List of commands
+                    Func<string, string> addGroup;
+                    Func<string, string, string> addGroupWithId;
+                    Func<string, string, string> addUserWithId;
+                    Func<string, string, string, string> addUserWithIdAndGroup;
+                    Func<string, string, string> addUserToGroup;
+
+                    try
+                    {
```

Comment:
exception handler seems broad, please narrow it exactly where InvalidOperation is expected to be thrown.  (probably the DockerExec)

Reply:
Fixed, thanks!